### PR TITLE
Add support for changing column types in MongoDB connector

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Add support for changing column types in MongoDB connector
Relates to #15515

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# MongoDB
* Add support for changing column types. ({issue}`15515`)
```

### CODE REVIEW

File: plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java [hunk 1]
```diff
@@ -254,6 +267,80 @@ public void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandl
...
+    {
+        MongoTableHandle table = (MongoTableHandle) tableHandle;
+        MongoColumnHandle column = (MongoColumnHandle) columnHandle;
+        if (!canSetColumnType(column.getType(), type)) {
+            throw new TrinoException(NOT_SUPPORTED, "Cannot change type from %s to %s".formatted(column.getType(), type));
+        }
+        mongoSession.setColumnType(table, column.getName(), type);
+    }
+
+    private static boolean canSetColumnType(Type sourceType, Type newType)
```
Comment (Praveen2112) [inline plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java:281]: Can we rename this method to `canAlterColumnType` ot `canChangeColumnType` ?

File: plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java [hunk 2]
```diff
@@ -810,6 +809,29 @@ protected void verifyTableNameLengthFailurePermissible(Throwable e)
...
     }
 
+    @Override
+    protected void verifySetColumnTypeFailurePermissible(Throwable e)
+    {
+        assertThat(e).hasMessageContaining("Cannot change type");
+    }
+
+    @Override
+    protected Optional<SetColumnTypeSetup> filterSetColumnTypesDataProvider(SetColumnTypeSetup setup)
```
Comment (Praveen2112) [inline plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java:819]: Do we additional tests for structural data types like array ?

Comment (Praveen2112) [reply plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java:819]: Specific to `rowType` where we could add a new field, drop a new field and rename a field ?

Comment (ebyhr) [reply plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java:819]: BaseConnectorTest already covers array and row types in `io.trino.testing.BaseConnectorTest#setColumnTypeSetupData`

Comment (Praveen2112) [reply plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java:819]: Oops I missed it. Thanks for pointing it out.

**Review by Praveen2112 — APPROVED** [review]

**Pull request merged**